### PR TITLE
feat: Add new points to the active spline

### LINKED ISSUE #26: Add new points to the active spline
**Describe the feature you'd like**
Add new points to the active spline with single clicks.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by SilviaZeta**
# Description

Adds the ability to add new points to the active spline, as described in #26. 

### How the feature is used:
After drawing a spline one can press the Enter key to switch from draw mode to edit mode (the user does not have to be aware of this distinction to be abled to use the tool). In edit mode, new points can be added with single clicks near the existing spline. When a new spline object is created or when a spline is deleted, mode is reset to draw.

closes #26

# Dependency changes

No

# Testing

N/A

# Documentation

N/A

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~I have made corresponding changes to the documentation~
- [x] My changes generate no new warnings
- [ ] ~I have added tests that prove my fix is effective or that my feature works~
- [ ] ~New and existing unit tests pass locally with my changes~

### CODE REVIEW

File: src/toolboxes/spline/Canvas.tsx [hunk 1]
```diff
@@ -331,15 +368,48 @@ export class SplineCanvas extends Component<Props> {
...
+        dist(x, y, prevPoint.x, prevPoint.y) +
+        dist(x, y, nextPoint.x, nextPoint.y);
+      // If the calculated distance is smaller than the min distance so far
+      if (minDist > newDist) {
+        // Update minimum distance and new point index
+        minDist = newDist;
+        newPointIndex = i;
+      }
+    }
+    coordinates.splice(newPointIndex, 0, { x: x, y: y }); // Add new point to the coordinates array
```
Comment (cooper667) [inline src/toolboxes/spline/Canvas.tsx:398]: For shorthand here you can replace `{x: x, y: y}` with `{x, y}` if you want

File: src/toolboxes/spline/Canvas.tsx [hunk 2]
```diff
@@ -331,15 +368,48 @@ export class SplineCanvas extends Component<Props> {
     );
   };
 
+  private addNewPointNearSpline = (x: number, y: number): void => {
+    // Add a new point near the spline.
+    const coordinates = this.props.annotationsObject.getActiveAnnotation()
```
Comment (cooper667) [inline src/toolboxes/spline/Canvas.tsx:373]: Replace with 
```const { coordinates } = this.props.annotationsObject.getActiveAnnotation();```

😄

**Review by cooper667 — APPROVED** [review]

File: src/keybindings.ts [hunk 3]
```diff
@@ -4,4 +4,6 @@ interface Keybindings {
 
 export const keybindings = {
   Backspace: "deleteSelectedPoint",
+  Enter: "changeSplineModeToEdit",
+  Escape: "removePointSelection",
```
Comment (ChasNelson1990) [inline src/keybindings.ts:8]: would this better with the name "deselectPoint"?

File: src/toolboxes/spline/Canvas.tsx [hunk 4]
```diff
@@ -38,47 +47,47 @@ export class SplineCanvas extends Component<Props> {
...
   drawSplineVector = (splineVector: XYPoint[], isActive = false): void => {
-    if (splineVector.length < 2) return;
+    if (splineVector.length === 0) return;
 
     const { canvasContext: context } = this.baseCanvas;
     const lineWidth = isActive ? 2 : 1;
-
-    // Clear the canvas
     context.lineWidth = lineWidth;
     context.strokeStyle = "#ff0000";
```
Comment (ChasNelson1990) [inline src/toolboxes/spline/Canvas.tsx:59]: Can we use the material UI colour names here? E.g. use the default primary and secondary colours?

**Review by ChasNelson1990 — APPROVED** [review]

**Review by ChrisBaidoo — APPROVED** [review]
LGTM

**Review by ChasNelson1990 — APPROVED** [review]

**Review by cooper667 — APPROVED** [review]

**Pull request merged**